src/pages/login/GitHub.tsx

```diff
@@ -0,0 +1,53 @@
+import * as pages from "codeforlife/components/page"
+import { type FC, useEffect } from "react"
+import CflLogoImage from "../../images/cfl_logo.png"
+import GitHubLoginButton from "../../components/GitHubLoginButton"
+import { Image } from "codeforlife/components"
+import { useLoginWithGitHubMutation } from "../../api/session"
+
+export interface GitHubProps {}
+
+const GitHub: FC<GitHubProps> = () => {
+  const [loginWithGitHub] = useLoginWithGitHubMutation()
+
+  useEffect(() => {
+    const urlParams = new URLSearchParams(window.location.search)
+    const code = urlParams.get("code")
+
+    if (code) {
+      urlParams.delete("code")
+      window.history.replaceState(null, "", window.location.pathname)
```

Comment:
you must use the `useNavigate` hook to navigate between pages. you can only navigate after a component has mounted.

```js
import { useNavigate } from "codeforlife/hooks/router"
...
const navigate = useNavigate()
...
useEffect(() => {
// navigate to the same page with "."
navigate(".", { replace: true })
}, [...])
```